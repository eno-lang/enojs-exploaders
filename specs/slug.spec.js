const eno = require('enojs');
const { slug } = require('../loaders.js');

const examples = {
  'eno-lang-article' : 'eno-lang-article',
  'eno_lang_article' : 'eno_lang_article',
  'eno-lang-article!': false,
  '%eno-lang-article': false,
  'eno lang article' : false,
  'enö-läng-ärticle' : false,
  'énó-láng-ártíclé' : false
};

describe('slug loader', () => {
  for(let [value, result] of Object.entries(examples)) {
    describe(value, () => {
      const document = eno.parse(`value: ${value}`);

      if(result === false) {
        it('throws an error', () => {
          expect(() => document.field('value', slug)).toThrowErrorMatchingSnapshot();
        });
      } else {
        it(`returns ${result}`, () => {
          expect(document.field('value', slug)).toBe(result);
        });
      }
    });
  }
});
