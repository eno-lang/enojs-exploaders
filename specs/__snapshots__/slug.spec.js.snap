// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`slug loader %eno-lang-article throws an error 1`] = `
"'value' must be a slug.

   Line | Content
 >    1 | value: %eno-lang-article
"
`;

exports[`slug loader eno lang article throws an error 1`] = `
"'value' must be a slug.

   Line | Content
 >    1 | value: eno lang article
"
`;

exports[`slug loader eno-lang-article! throws an error 1`] = `
"'value' must be a slug.

   Line | Content
 >    1 | value: eno-lang-article!
"
`;

exports[`slug loader enö-läng-ärticle throws an error 1`] = `
"'value' must be a slug.

   Line | Content
 >    1 | value: enö-läng-ärticle
"
`;

exports[`slug loader énó-láng-ártíclé throws an error 1`] = `
"'value' must be a slug.

   Line | Content
 >    1 | value: énó-láng-ártíclé
"
`;
